src/FSharpVSPowerTools.Logic.VS2015/PeekDefinition.fs

```diff
@@ -0,0 +1,136 @@
+                                                span.StartColumn,
+                                                true)
+
+                        resultCollection.Add result
+    //                callback.ReportProgress(100 * ++processedSourceLocations / sourceLocations.Count);
+            }
+
+type PeekableItemSource
+    (
+        buffer: ITextBuffer,
+        doc: ITextDocument,
+        peekResultFactory: IPeekResultFactory,
+        serviceProvider: System.IServiceProvider,
+        projectFactory: ProjectFactory,
+        vsLanguageService: VSLanguageService
+    ) =
+
+    let getCurrentFilePathProjectAndDoc () =
+        maybe {
+            let filepath = doc.FilePath
+            let dte = serviceProvider.GetService<EnvDTE.DTE, SDTE>()
+            let! doc = dte.GetCurrentDocument(filepath)
+            let! project = projectFactory.CreateForDocument buffer doc
+            return filepath, project, doc
+        }
+
+    let getDefinitionRange (point: SnapshotPoint) =
+        async {
+            let projectItems = 
+                maybe {
+                    let! _, project, doc = getCurrentFilePathProjectAndDoc()
+                    let! span, symbol = vsLanguageService.GetSymbol(point, doc.FullName, project)
+                    return doc.FullName, project, span, symbol 
+                }
+                 
+            match projectItems with
+            | Some (file, project, span, symbol) ->
+                let! symbolUse = vsLanguageService.GetFSharpSymbolUse(span, symbol, file, project, AllowStaleResults.MatchingSource)
+                match symbolUse with
+                | Some (_fsSymbolUse, fileScopedCheckResults) ->
+                    let start = span.Start
+                    let lineStr = start.GetContainingLine().GetText()
+                    let! findDeclResult = fileScopedCheckResults.GetDeclarationLocation(symbol.Line, symbol.RightColumn, lineStr, symbol.Text, preferSignature=false)
+                    return
+                        match findDeclResult with
+                        | FSharpFindDeclResult.DeclFound range -> Some (span, range)
+                        | _ -> None
+                | _ -> return None
+            | _ -> return None
+        }
+
+    interface IPeekableItemSource with
+        member __.AugmentPeekSession (session, peekableItems) =
+            asyncMaybe {
+                do! if String.Equals(session.RelationshipName, PredefinedPeekRelationships.Definitions.Name, 
+                                     StringComparison.OrdinalIgnoreCase) then Some() else None
+                let! triggerPoint = session.GetTriggerPoint(buffer.CurrentSnapshot) |> Option.ofNullable
+                let! symbolSpan, definitionRange = getDefinitionRange triggerPoint
+                peekableItems.Add (DefinitionPeekableItem(symbolSpan, definitionRange, peekResultFactory))
+            } |> Async.RunSynchronously |> ignore
```

Comment:
Could we use `Async.StartInThreadPoolSafe` or `Async.StartImmediateSafe` here instead?